src/rt/dwarfeh.d

```diff
@@ -780,3 +807,99 @@ void terminate(uint line) @nogc
     printf("dwarfeh(%u) fatal error\n", line);
     abort();     // unceremoniously exit
 }
+
+
+/****************************** C++ Support *****************************/
+
+enum _Unwind_Exception_Class cppExceptionClass =
+        (cast(_Unwind_Exception_Class)'G' << 56) |
+        (cast(_Unwind_Exception_Class)'N' << 48) |
+        (cast(_Unwind_Exception_Class)'U' << 40) |
+        (cast(_Unwind_Exception_Class)'C' << 32) |
+        (cast(_Unwind_Exception_Class)'C' << 24) |
+        (cast(_Unwind_Exception_Class)'+' << 16) |
+        (cast(_Unwind_Exception_Class)'+' <<  8) |
+        (cast(_Unwind_Exception_Class)0 <<  0);
+
+enum _Unwind_Exception_Class cppExceptionClass1 = cppExceptionClass + 1;
+
+
+/*****************************************
+ * Get Pointer to Thrown Object if type of thrown object is implicitly
+ * convertible to the catch type.
+ * Params:
+ *      exceptionObject = language specific exception information
+ *      sti = type of catch clause
+ * Returns:
+ *      null if not caught, pointer to thrown object if caught
+ */
+void* getCppPtrToThrownObject(_Unwind_Exception* exceptionObject, CppTypeInfo sti)
+{
+    void* p;    // pointer to thrown object
+    if (exceptionObject.exception_class & 1)
+        p = CppExceptionHeader.toExceptionHeader(exceptionObject).ptr;
+    else
+        p = cast(void*)(exceptionObject + 1);           // thrown object is immediately after it
+
+    const tt = (cast(CppExceptionHeader*)p - 1).typeinfo;
+
+    if (tt.__is_pointer_p())
+        p = *cast(void**)p;
+
+    // Pointer adjustment may be necessary due to multiple inheritance
+    return (sti.__do_catch(tt, &p, 1)) ? p : null;
```

Comment:
`operator==` in my libstdc++ looks like
```
bool std::type_info::
operator== (const std::type_info& arg) const
{
  return (&arg == this) || (__builtin_strcmp (name (), arg.name ()) == 0);
}
```
However in other versions it may differ.